### PR TITLE
Add Chrome MV3 support for user allowlisting

### PR DESCRIPTION
**Pull request by kzar**
Users can allowlist (disable protections) for websites from the popup
UI and also from the options page. To make that work with Chrome MV3,
we need to add a declarativeNetRequest rule that ensures requests
initiated by domains that the user has allowlisted aren't blocked or
redirected.

**Reviewer:** @jdorweiler 
**CC:** @jonathanKingston 

## Steps to test this PR:
1. Produce a Chrome MV3 build of the extension (`npm install && npm run dev-chrome-mv3`).
2. Install that in Chrome.
3. Try allowlisting websites, both from the options page and the popup UI and ensure it works.
4. Click to reload the extension from the extensions page.
5. Make sure the websites you allowlisted are still allowlisted (try opening them in a new tab too).
6. Remove any allowlisted websites from the options page and ensure they are no longer allowlisted.

## Automated tests:
- [x] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

**Comment by jdorweiler** [comment]
I'm getting the same startup error I was seeing in #1450 when I tried to include the declarative net module. 
Screen Shot 2022-10-25 at 7 57 21 AM

**Comment by kzar** [comment]
> I'm getting the same startup error I was seeing in #1450 when I tried to include the declarative net module. Screen Shot 2022-10-25 at 7 57 21 AM

@jdorweiler Dang, yea I have also seen that one cropping up. I'm going to file a follow-up task to investigate, it seems only tangentially (if at all) related to these changes.

**Comment by jonathanKingston** [comment]
> > I'm getting the same startup error I was seeing in #1450 when I tried to include the declarative net module. Screen Shot 2022-10-25 at 7 57 21 AM
> 
> @jdorweiler Dang, yea I have also seen that one cropping up. I'm going to file a follow-up task to investigate, it seems only tangentially (if at all) related to these changes.

I've still yet to have seen this; please create an issue with clear reproduction steps. It's clearly unrelated.

File: shared/js/background/startup.es6.js [hunk 1]
```diff
@@ -12,11 +12,9 @@ const tdsStorage = require('./storage/tds.es6')
 const trackers = require('./trackers.es6')
 const dnrSessionId = require('./dnr-session-rule-id')
 const { fetchAlias, showContextMenuAction } = require('./email-utils.es6')
-const manifestVersion = browserWrapper.getManifestVersion()
-/** @module */
+const { startupReadyResolve } = require('./ready.es6')
```
Comment (jonathanKingston) [inline shared/js/background/startup.es6.js:15]: We can drop the .es6 from this file; it's only used when transpilation is needed which isn't the case ever.

**Comment by jdorweiler** [comment]
@kzar looks good, I think you're just missing the page reload when toggling

**Review by jdorweiler — APPROVED** [review]

**Pull request merged**